app/src/main/java/com/hedvig/app/feature/offer/usecase/datacollectionresult/DataCollectionResult.kt

```diff
@@ -4,18 +4,12 @@ import com.hedvig.android.owldroid.graphql.DataCollectionResultQuery
 import com.hedvig.app.util.apollo.toMonetaryAmount
 import javax.money.MonetaryAmount
 
-data class DataCollectionResult(
-    val collectedList: List<CollectedInsuranceData>,
-) {
-    companion object {
-        fun fromDto(dto: DataCollectionResultQuery.Data): DataCollectionResult {
-            return DataCollectionResult(
-                dto.externalInsuranceProvider
-                    ?.dataCollectionV2
-                    ?.mapNotNull(CollectedInsuranceData::fromDto) ?: emptyList()
-            )
-        }
-    }
+sealed class DataCollectionResult {
+
+    object Empty : DataCollectionResult()
+    data class Content(
+        val collectedList: List<CollectedInsuranceData>,
```

Comment:
It'd be nice to have a NonEmptyList here instead right :D

Reply:
Yep!